scripts/add_cluster_resource.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/sh
+
+set -o errexit
+trap 'echo "Aborting due to errexit on line $LINENO. Exit code: $?" >&2' ERR
+set -o errtrace
+set -o pipefail
+
+_BN="$(basename "${0}")"
+
+_usage() {
+cat <<HEREDOC
+
+Fetch a cluster resource from github and add it to cluster-scope app.
+This script requires the user to be logged into an existing OCP cluster
+that supports the api resources meaning to be added.
+
+Usage:
+  ${_BN} [env] [cluster] [flags]
+
+Mandatory flags (mutually exclusive):
+  -f, --file    path to file on local filesystem
+  -u, --url     remote git url for the raw file
+
+Optional Flags:
+  -h, --help    help for this script
+
+Example:
+  ${_BN} moc zero -f /tmp/manifest.yaml
+  ${_BN} moc zero -u https://raw.githubusercontent.com/<github_user>/<repo>/master/<path-to-manifest>/manifest.yaml
+
+HEREDOC
+}
+
+# Keep values sorted alphabetically since `kustomize edit resource add` just appends.
+_sort_kustomize_values(){
```

Comment:
`kustomize edit add ....` doesn't care about alphabetical sorting, so this is a potentially convoluted way of ensuring it stays sorted, open to better methods.

Reply:
This is really beginning to smell like "we should write this in a language with native support for YAML".

Reply:
I don't have an better idea. We're sorting arrays here.. which are usually ordered :confused: I agree that we might want to start the golang cli finally. :smile:

Reply:
I am not against doing this

Reply:
totally agree